docs/framework/wcf/samples/custom-lifetime.md

```diff
@@ -1,219 +1,227 @@
-           Collection<DispatchBehavior> behaviors,  
-           Collection<BindingParameterCollection> parameters)  
-{  
-    CustomLifetimeLease customLease = new CustomLifetimeLease();  
-    CustomLeaseInitializer initializer =   
-                new CustomLeaseInitializer(timeout);  
-  
-    foreach (DispatchBehavior dispatchBehavior in behaviors)  
-    {  
-        dispatchBehavior.InstanceContextLifetimes.Add(customLease);  
-        dispatchBehavior.InstanceContextInitializers.Add(initializer);  
-    }  
-}  
-```  
-  
- This behavior can be added to a sample service class by annotating it with the `CustomLeaseTime` attribute.  
-  
-```  
-[ServiceBehavior(InstanceContextMode=InstanceContextMode.Shareable)]  
-[CustomLeaseTime(Timeout = 20000)]  
-public class EchoService : IEchoService  
-{  
-  //…  
-}  
-```  
-  
- When you run the sample, the operation requests and responses are displayed in both the service and client console windows. Press ENTER in each console window to shut down the service and client.  
-  
-#### To set up, build, and run the sample  
-  
-1.  Ensure that you have performed the [One-Time Setup Procedure for the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/one-time-setup-procedure-for-the-wcf-samples.md).  
-  
-2.  To build the C# or Visual Basic .NET edition of the solution, follow the instructions in [Building the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/building-the-samples.md).  
-  
-3.  To run the sample in a single- or cross-machine configuration, follow the instructions in [Running the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/running-the-samples.md).  
-  
+> Using thread locking is not a recommended approach because serialization can severely affect the performance of your application.
+
+A private member variable is used in the `CustomLifetimeLease` class to track the `IsIdle` value. Each time the value of <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> is retrieved, the `IsIdle` private member is returned and reset to `false`. It is essential to set this value to `false` in order to make sure the Dispatcher calls the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.NotifyIdle%2A> method.
+
+```csharp
+public bool IsIdle(InstanceContext instanceContext)
+{
+    get
+    {
+        lock (thisLock)
+        {
+            //...
+            bool idleCopy = isIdle;
+            isIdle = false;
+            return idleCopy;
+        }
+    }
+}
+```
+
+If the `IInstanceContextProvider.IsIdle` property returns `false`, the Dispatcher registers a callback function by using the `NotifyIdle` method. This method receives a reference to the <xref:System.ServiceModel.InstanceContext> being released. Therefore, the sample code can query the `ICustomLease` type extension and check the `ICustomLease.IsIdle` property in the extended state.
```

Comment:
Should xref be used for IsIdle and NotifyIdle? Also, a few corrections property>method, it should read:

If the `IInstanceContextProvider.IsIdle` method returns `false`, the Dispatcher registers a callback function by using the `NotifyIdle` method. This method receives a reference to the <xref:System.ServiceModel.InstanceContext> being released. Therefore, the sample code can query the `ICustomLease` type extension and check the `ICustomLease.IsIdle` property in the extended state.